zcash_client_sqlite/src/wallet/init/migrations/v_transactions_net.rs

```diff
@@ -0,0 +1,318 @@
+//! Migration that fixes a bug in v_transactions that caused the change to be incorrectly ignored
+//! as received value.
+use std::collections::HashSet;
+
+use rusqlite;
+use schemer;
+use schemer_rusqlite::RusqliteMigration;
+use uuid::Uuid;
+
+use super::add_transaction_views;
+use crate::wallet::init::WalletMigrationError;
+
+pub(super) const MIGRATION_ID: Uuid = Uuid::from_fields(
+    0x2aa4d24f,
+    0x51aa,
+    0x4a4c,
+    b"\x8d\x9b\xe5\xb8\xa7\x62\x86\x5f",
+);
+
+pub(crate) struct Migration;
+
+impl schemer::Migration for Migration {
+    fn id(&self) -> Uuid {
+        MIGRATION_ID
+    }
+
+    fn dependencies(&self) -> HashSet<Uuid> {
+        [add_transaction_views::MIGRATION_ID].into_iter().collect()
+    }
+
+    fn description(&self) -> &'static str {
+        "Fix transaction views to correctly handle double-entry accounting for change."
+    }
+}
+
+impl RusqliteMigration for Migration {
+    type Error = WalletMigrationError;
+
+    fn up(&self, transaction: &rusqlite::Transaction) -> Result<(), WalletMigrationError> {
+        transaction.execute_batch(
+            "INSERT INTO sent_notes (tx, output_pool, output_index, from_account, to_account, value)
+             SELECT tx, 2, output_index, account, account, value 
+             FROM received_notes
+             WHERE received_notes.is_change
+             EXCEPT
+             SELECT tx, 2, output_index, from_account, from_account, value
+             FROM sent_notes")?;
```

Comment:
What is this `EXCEPT` necessary for? Do we have a test that exercises it?

Reply:
This EXCEPT is the critical part that ensures we don't create redundant `sent_notes` entries.

Reply:
Ah, I see. The `from_account, from_account` part was what confused me; it seems overly restrictive since we know that the relevant row will have `(tx, 2, output_index)` be unique (so that is what we really want to match on). If we need to select values to have this side be compatible with the `EXCEPT`, we should select whatever is in the `to_account` column rather than selecting `from_account` twice.